### DISCUSSION TITLE
Error		This version of WindowsAppSDK requires .NET 6+ and WinRT.Runtime.dll version 2.0 or greater.

### DISCUSSION THREAD

**Discussion by sam-wheat** [unoplatform/uno#10649] [Q&A] [answered]
Updating Microsoft.WindowsAppSDK from 1.1.5 to 1.2.221116.1  produces this error:

> Error		This version of WindowsAppSDK requires .NET 6+ and WinRT.Runtime.dll version 2.0 or greater.
    Please update to .NET SDK 6.0.109, 6.0.304 or 6.0.401 (or later).
    Or add a temporary Microsoft.Windows.SDK.NET.Ref reference of version 10.0.*.27 or later.
    For example,
   
        <PropertyGroup>
            <WindowsSdkPackageVersion>10.0.<Target Windows SDK Build Number>.27</WindowsSdkPackageVersion>
        </PropertyGroup>
	

I found [this issue](https://github.com/microsoft/WindowsAppSDK/issues/2439) which reccommends  this change:

	<WindowsSdkPackageVersion>10.0.XYZ.23</WindowsSdkPackageVersion>
	to
	<WindowsSdkPackageVersion>10.0.XYZ.24</WindowsSdkPackageVersion>
	
I don't think that applies to me since `WindowsSdkPackageVersion` is not specified in my .csproj.

.csproj:

	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net7.0-windows10.0.19041.0</TargetFramework>
		<TargetPlatformMinVersion>10.0.18362.0</TargetPlatformMinVersion>
		<ApplicationManifest>app.manifest</ApplicationManifest>
		<Platforms>x86;x64;arm64</Platforms>
		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
		<PublishProfile>win10-$(Platform).pubxml</PublishProfile>
		<UseWinUI>true</UseWinUI>
		<EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
	</PropertyGroup>

C:\Users\sam>dotnet --info
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

Host:
  Version:      7.0.0
  Architecture: x64
  Commit:       d099f075e4

.NET SDKs installed:

  7.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:

  Microsoft.AspNetCore.All 2.2.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]

  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]

  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  arm64 [C:\Program Files\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\arm64\InstallLocation]
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. Can you take a look at [this section in your project](https://github.com/unoplatform/uno/blob/master/src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/WinUI/UnoQuickStart.Windows.csproj#L41-L53)? You should be able to adjust the versions for the build to work.

**Comment by ttustonic** [comment]
Link in the answer seems to be old.
This one should work
https://github.com/unoplatform/uno/blob/master/src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Windows/UnoQuickStart.Windows.csproj#L41-L53

**Comment by gabriel-vanca** [comment]
Still having this issue.

**Reply by jeromelaban** [reply]
Thanks for the report. This is not an issue that is caused by Uno and is caused by updating to WinAppSDK 1.6, and you can follow the directions provided by the error message. 

We will fix this error by setting the version in the template's csproj temporarily, but it will not be needed once newer versions of .NET get released.
